src/scalarstats.jl

```diff
@@ -584,21 +584,12 @@ end
 Compute the cross entropy between `p` and `q`, optionally specifying a real
 number `b` such that the result is scaled by `1/log(b)`.
 """
-function crossentropy(p::AbstractArray{T}, q::AbstractArray{T}) where T<:Real
+function crossentropy(p::AbstractArray{<:Real}, q::AbstractArray{<:Real})
     length(p) == length(q) || throw(DimensionMismatch("Inconsistent array length."))
-    s = 0.
-    z = zero(T)
-    for i = 1:length(p)
-        @inbounds pi = p[i]
-        @inbounds qi = q[i]
-        if pi > z
-            s += pi * log(qi)
-        end
-    end
-    return -s
+    return - sum(xlogy(pi, qi) for (pi, qi) in zip(p, q))
```

Comment:
Note that using lazy broadcasting (on recent julia versions) is more efficient and has better precision as it uses pairwise summation. Maybe that would fix the performance regression?
```julia
    return - sum(Broadcast.instantiate(Broadcast.broadcasted(xlogx, p, q)))
```

Reply:
I've seen the Julia PR and played around with `sum(Brodcast....)` in some local branch of Distributions but in my benchmarks there another implementation was actually faster. I don't remember the details right now and I assume it was not really comparable to this example here though. I'll check if it helps but I guess we have to live with a slight performance regression - the existing implementations are not type stable and less general, so it is not completely fair to compare the performance of the bugfixes in this PR to the performance of the incorrect existing implementation (of course, we should still try to improve the performance of the bugfixes as much as possible).

Reply:
Improved precision may still be valuable anyway.